### PR TITLE
started refactoring code and fixing bugs

### PR DESCRIPTION
**Pull request by vladciuca**

### CODE REVIEW

File: src/pages/CoinList/index.js [hunk 1]
```diff
@@ -75,13 +68,28 @@ class CoinList extends React.Component {
...
+        if (Object.values(storageFavoriteList).length === 0) return;
+        const favoriteList = Object.values(storageFavoriteList).reduce(
+          (acc, current, index, array) => {
+            if (index === array.length - 1) {
+              return acc + `${current}`;
+            }
+            return acc + `${current}%2C%20`;
+          },
+          ""
+        );
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinList/index.js]: you cannot use query-string to parse this?

File: src/pages/CoinList/index.js [hunk 2]
```diff
@@ -275,14 +309,15 @@ class CoinList extends React.Component {
...
-          <div>
-            <SkeletonCoinList coinsPerPage={this.state.coinsPerPage} />
-          </div>
-        )}
+        {this.state.isLoading &&
+          (this.state.showFavorites && favoriteCoinsLength() < 1 ? (
+            <SkeletonCoinList coinsPerPage={favoriteCoinsLength()} />
+          ) : (
+            <SkeletonCoinList coinsPerPage={this.getScreenWidth()} />
+          ))}
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinList/index.js]: this block looks a bit messy, can you make a few variables out of all these conditions?

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -10,15 +10,16 @@
     "axios": "^0.21.1",
     "chart.js": "^2.9.4",
     "default": "^0.1.1",
+    "dominant-color": "0.0.1",
```
Comment (yoloOnTheBattlefield) [inline package.json]: you are not using this, uninstall

File: src/App.js [hunk 4]
```diff
@@ -1,53 +1,105 @@
...
     const currency = e.target.value;
+    storage("set", "currency", currency);
     this.setState({ currency });
-    localStorage.setItem("currency", JSON.stringify(currency));
+  };
+  handleHomeLink = (link) => {
+    this.setState({ homePageLink: link });
   };
   componentDidMount() {
+    console.log("IN LOCAL STORAGE", storage("get", "theme"));
```
Comment (yoloOnTheBattlefield) [inline src/App.js]: 😅

File: src/App.js [hunk 5]
```diff
@@ -1,53 +1,105 @@
...
     this.setState({
-      currency: JSON.parse(localStorage.getItem("currency")) || "usd",
+      theme: storage("get", "theme") || true,
+      currency: storage("get", "currency") || "usd",
     });
   }
   render() {
-    const { currency } = this.state;
+    console.log("IN STATE", this.state.theme);
```
Comment (yoloOnTheBattlefield) [inline src/App.js]: 😅

File: src/GlobalStyle.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+::-webkit-scrollbar-thumb {
+  background: ${(props) => props.theme.secondary};
+  border-radius: 10px;
+}
+::-webkit-scrollbar-thumb:hover {
+  background: ${(props) => props.theme.primary};
+}
+.App {
+  text-align: center;
+}
```
Comment (yoloOnTheBattlefield) [inline src/GlobalStyle.js]: delete .App

File: src/components/CoinListHeader/CoinListHeader.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  const priceChangeSorts = [
+    { title: "1h%", value: "priceChangePercentage1hInCurrency" },
+    { title: "24h%", value: "priceChangePercentage24hInCurrency" },
+    { title: "7d%", value: "priceChangePercentage7dInCurrency" },
+  ];
+  const categoryChange = (category) => {
+    if (category === "decentralizedFinanceDefi" || category === "stablecoins") {
+      return true;
+    }
+  };
```
Comment (yoloOnTheBattlefield) [inline src/components/CoinListHeader/CoinListHeader.js]: `const categoryChange = (category) => category === "decentralizedFinanceDefi" || category === "stablecoins"`

File: src/components/CoinListItem/CoinListItem.js [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+              <Tooltip
+                placement="bottom"
+                title={`${currencySymbol}${formatNumber(totalVolume)}`}
+              >
+                {currencySymbol}
+                {convertLongNumber(totalVolume)}
+              </Tooltip>
+            </span>
+          )}
+        </div>
```
Comment (yoloOnTheBattlefield) [inline src/components/CoinListItem/CoinListItem.js]: can you make this a reusable component?

File: src/pages/CoinPage/CoinPage.js [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+import { Background, Card } from "./CoinPage.styles";
+
+export default class CoinPage extends React.Component {
+  state = {
+    coinData: null,
+  };
+  getCoin = async () => {
+    try {
+      this.setState({ isLoading: true });
+      const coinId = this.props.match.params.id;
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinPage/CoinPage.js]: maybe change in App where you render coin page, id to coinId

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/components/GlobalData/GlobalData.js [hunk 10]
```diff
@@ -53,6 +59,7 @@ export default class GlobalData extends React.Component {
   render() {
     const hasData = !this.state.isLoading && this.state.globalData;
     const globalData = this.state.globalData;
```
Comment (redliz5808) [inline src/components/GlobalData/GlobalData.js]: could do `const { globalData } = this.state` here.

**Review by redliz5808 — COMMENTED** [review]
Man I am so impressed by your app every time I see it. Keep up the good work!

File: src/pages/CoinList/index.js (hunk 1 above)
Comment (vladciuca) [reply src/pages/CoinList/index.js]: I tryed it and I cant get it to return the proper format

**Pull request merged**